src/main/kotlin/com/jetbrains/snakecharm/lang/psi/SMKParamsReference.kt

```diff
@@ -36,7 +36,7 @@ class SMKParamsReference(
         return variants.toTypedArray()
     }
 
-    private fun getParamsSection(): SMKRuleParameterListStatement? {
+    private fun getParamsSection(): SmkSectionStatement? {
```

Comment:
SmkRule indeed can contain only `SMKRuleParameterListStatement`. May be we need generic type attribute in SmkRuleLike, which specifies sub-type of SmkSectionStatement?